rclcpp_components/src/node_main.cpp.in

```diff
@@ -62,6 +62,15 @@ int main(int argc, char * argv[])
     }
   }
 
+  if (exec.get_all_callback_groups().empty()) {
+    RCLCPP_ERROR(
+      logger,
+      "Class %s not found in library %s",
+      class_name.c_str(),
+      library_name.c_str());
+    return 1;
+  }
+
```

Comment:
We discussed this in one of the ROS 2 core meetings, and we think this test is actually wrong.  It should be permissible to load in components that have no callback groups, i.e. it is a design goal to allow components with no nodes.

Instead, this should probably change to be a test to see if there were zero classes.  If that was the case, then we should definitely fail, since we failed to do what the user asked us to do.

Reply:
That sounds reasonable, although I'm not sure I understand the cases where you would load a class that doesn't add any callback groups to the executor. If there are no callback groups, won't the component hang forever and do nothing?

Reply:
> That sounds reasonable, although I'm not sure I understand the cases where you would load a class that doesn't add any callback groups to the executor. If there are no callback groups, won't the component hang forever and do nothing?

Not necessarily.  The idea of a component is a generic concept, not directly tied to ROS 2 at all.  At the end of the day, it is just an entry point that we call in a shared object.  So if your component comes up and starts a thread that does processing of some sort, that will still happen.  Of course, it can't really interact with the ROS 2 system until the executor starts, but that is a problem for the component developer.

Reply:
Ok @clalancette I have made this change, so it simply checks if no class was found.